docs/package_publishing.md

```diff
@@ -0,0 +1,73 @@
+framework. That package, that we know as component, contains not only the code but also its configuration and dependencies (```component.yaml``` in the image). Once the package is completed,
+it needs to be pushed to IPFS registry so other people can access to it. Using the ```autonomy``` CLI a developer can perform this action, and an IPFS hash that
+points to this specific version of the code will be returned.
+
+<figure markdown>
+![](./images/create_component.svg)
+<figcaption>A developer creates a component and pushes it to the IPFS registry</figcaption>
+</figure>
+
+## Registering a component
+
+Once the code has been tested and considered final, it is time to register the component in the on-chain protocol so other people can find it and reference it. Using the protocol frontend, the developer must fill a form with
+the details of the package, like its IPFS hash, its owner address, description and so on. Once every detail has been filled out, the frontend will generate a ```json```
+file with all the required metadata, push it to IPFS and retrieve the corresponding hash. After this, the developer can register that meta-data hash on the on-chain protocol and the component
+owner will receive in his wallet an NFT that represents the ownsership of this component.
+
+It is important to emphasize that two different pushes to IPFS have been
+performed up to this point: the first one, done by the developer to push the code and a second one done through the frontend to push the metadata (that contains the code hash itself).
+
+<figure markdown>
+![](./images/register_component.svg)
+<figcaption>A developer registers a component into the protocol</figcaption>
+</figure>
+
+## Registering an agent
+
+At some point, a service owner might want to create an agent that uses the previously registered component analogously to how some programs use other libraries. In an similar way to how a component was registered, this service owner
+registers an agent, specifying all the components that make up the agent. A NFT will be minted as well and sent to the owner wallet.
+
+Note that this agent is not a running agent, but a definition of an agent that node operators will later use to run to deploy decentralized services.
+
+<figure markdown>
+![](./images/register_agent.svg)
+<figcaption>An operator registers an agent into the protocol</figcaption>
+</figure>
+
+## Registering a service
+
+After the agent is registered, a service based on it can be registered as well. Services are composed of one or more agents that can be run by multiple agent instances themselves: in the most simple case, a service could be composed of a single agent run by a single agent instance, though this would mean that the service would be completely centralized and not fault-tolerant. Therefore, it is expected that each agent in a service is run by multiple agent instances so the service benefits from those features.
```

Comment:
```suggestion
After the agent is registered, a service based on it can be registered as well. Services are composed of one or more (canonical) agents. The service then specifies how many agent instances are required of each (canonical) agent. In the most simple case, a service could be composed of a single agent run as a single agent instance. However, this would mean that the service would be completely centralized and not fault-tolerant. Therefore, it is common that each agent in a service is run by multiple agent instances.
```